src/messages/mod.rs

```diff
@@ -189,13 +189,23 @@ impl HopMessage {
     /// This does not imply that the message came from a known node. That requires a check against
     /// the routing table to identify the name associated with the `verification_key`.
     pub fn verify(&self, verification_key: &PublicSignKey) -> Result<()> {
-        let signed_bytes = serialise(&self.content)?;
+        let signed_bytes = HopMessage::serialise_content(&self.content)?;
         if verification_key.verify_detached(&self.signature, &signed_bytes) {
             Ok(())
         } else {
             Err(RoutingError::FailedSignature)
         }
     }
+
+    fn serialise_content(_content: &SignedMessage) -> Result<Vec<u8>> {
+        #[cfg(not(feature = "mock_serialise_send"))]
+        let result = serialise(_content)?;
+
+        #[cfg(feature = "mock_serialise_send")]
+        let result = b"HopMessage.content".to_vec();
```

Comment:
Even with mock serialization and mock cryptography, wouldn't we still want to distinguish between signatures of different messages? (But maybe not; the huge speedup may be worth it.)

Reply:
If there is a bug like that, I assume test without mock_serialisation_send would catch it super quick (probably in CI). I think it serve a different purpose, so I do not believe this will be an issue.